src/hyperdeck/response-handler.js

```diff
@@ -12,73 +12,78 @@ var ASYNC_RESPONSE_REGEX = /^(5[0-9]{2} )/;
 function ResponseHandler(clientSocket) {
   var destroyed = false;
   var notifier = new events.EventEmitter();
-  var buffer = "";
-  var nextResponseIsMultiline = false;
-  
-  function isBufferComplete() {
-    var lines = buffer.split("\r\n");
-    // there will always be an empty element at the end as every line will always end in "\r\n"
-    // so remove it.
-    // i.e. "1\r\n2\r\n" => ["1", "2", ""]
-    lines.pop();
-    if (lines.length === 1) {
-      var asyncResponse = ASYNC_RESPONSE_REGEX.exec(lines[0]);
-      if (!asyncResponse) {
-        if (nextResponseIsMultiline) {
-          logger.debug('Assuming buffer is not complete because nextResponseIsMultiline flag is set.');
-          nextResponseIsMultiline = false;
-          return false;
-        }
-      }
-      // is it a single line response?
-      return lines[0].indexOf(":") !== lines[0].length-1;
-    } 
-    // multi line response so waiting for a blank line to signify end
-    return lines[lines.length-1] === "";
+  var buffer = [];
+
+  function isRespComplete() {
+    let complete = false;
+
+    if (buffer.length === 1) {
+      // a single line response
+      // single line response first line should never ends with ":" and
+      // 1XX and 200 is always single line response
+      complete = buffer[0].search(/^1\d\d\s|200\sok$/) === 0 && buffer[0].endsWith(':') === false;
+    } else if (buffer.length >= 3) {
```

Comment:
why `>= 3`?

Reply:
multiline response end will be always terminate on 3rd or larger index (if not then its single line response)

Line 1 data:
Line 2
""  ← termination

Reply:
I see. I think it's safer just to wait for an empty line though so if for some random reason we got
```
Line 1 data:
"" ← termination
```
we wouldn't end up broken